alacritty/src/ipc.rs

```diff
@@ -40,13 +45,39 @@ pub fn spawn_ipc_socket(options: &Options, event_proxy: EventLoopProxy<Event>) -
 
     // Spawn a thread to listen on the IPC socket.
     thread::spawn_named("socket listener", move || {
-        // Accept up to 2 bytes to ensure only one byte is received.
-        // This ensures forward-compatibility.
-        let mut buf = [0; 2];
-
-        while let Ok(received) = socket.recv(&mut buf) {
-            if buf[..received] == SOCKET_MESSAGE_CREATE_WINDOW {
-                let _ = event_proxy.send_event(Event::new(EventType::CreateWindow, None));
+        let mut data = Vec::new();
+        for stream in listener.incoming() {
+            let stream = match stream {
+                Ok(stream) => stream,
+                Err(err) => {
+                    warn!("Failed to get incomming stream! {}", err);
+                    continue;
+                },
+            };
+
+            data.clear();
+            let mut stream = BufReader::new(stream);
+
+            match stream.read_until(JSON_SEPARATOR, &mut data) {
+                Ok(0) | Err(_) => continue,
+                Ok(bytes_read) => bytes_read,
+            };
```

Comment:
I don't think `\n` as `JSON_SEPARATOR` makes a lot of sense. Probably don't need any constant at all but if we do it should probably be `MESSAGE_SEPARATOR` or something since it's not actually related to the format at all.

More importantly though, this method looks a lot like `read_line`. Do we really need to store things in a `Vec` for serde, can't we just use a `String` instead and sidestep this issue of actually passing a separator?

Reply:
I wasn't entirely sure how data is separated on the wire here, so I just wrote it in a way I'd know it'll work for sure. Will try to optimize it.